src/Compilers/CSharp/Test/Semantic/Semantics/StaticNullChecking_FlowAnalysis.cs

```diff
@@ -1282,6 +1296,349 @@ static void G()
             comp.VerifyDiagnostics();
         }
 
+        [Fact]
+        public void IdentityConversion_NullCoalescingOperator_01()
+        {
+            var source =
+@"interface I<T> { }
+interface IIn<in T> { }
+interface IOut<out T> { }
+class C
+{
+    static void F(I<object>? x, I<object?> y)
+    {
+        I<object> z = x ?? y;
```

Comment:
One of the `z`s doesn't have a warning, but I don't know which.
Consider numbering the `z`s and `w`s for readability, or putting comments `// warning 1` (etc).